vars/junit.groovy

```diff
@@ -0,0 +1,51 @@
+// Licensed to Elasticsearch B.V. under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Elasticsearch B.V. licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+/**
+Wrap the junit built-in step to perform extra operations with the test reports.
+
+    pipeline {
+        ...
+        stages {
+            stage(...) {
+                post {
+                    always {
+                        // JUnit on steroids
+                        junit(callback: closure(), ...)
+                    }
+                }
+            }
+        }
+        ...
+    }
+*/
+def call(Map args = [:]) {
+  def hasCallback = args.containsKey('callback') ? true : false
+  def testResults = args.containsKey('testResults') ? args.testResults : error('junit: testResults parameter is required')
```

Comment:
I'd say to delegate this to the `junit` step.

```suggestion
```

Reply:
I copy&paste from the other key, and then refactored for simplification, forgetting to update the ternary operator, thanks!

Reply:
I copied this assertion from the `junitAndStore` https://github.com/elastic/apm-pipeline-library/blob/master/vars/junitAndStore.groovy#L43

Reply:
`junitAndStore ` is a bit opinionated, since it uses `stash` internally. In this case, it's needed for the `stash`. 
While in here, It's not used outside of the `junit` scope

Reply:
Makes sense. Will remove this verification, as we can re-add it whenever we use it